src/openrct2/rct1/S4Importer.cpp

```diff
@@ -2940,6 +2943,50 @@ class S4Importer final : public IParkImporter
         }
     }
 
+    /**
+     * In Urban Park, the entrance and exit of the merry-go-round are the wrong way round. This code fixes that.
+     * To avoid messing up saves (in which this problem is most likely solved by the user), only carry out this
+     * fix when loading from a scenario.
+     */
+    void FixUrbanPark()
+    {
+        if (_s4.scenario_slot_index == SC_URBAN_PARK && _isScenario)
+        {
+            // First, make the queuing peep exit
+            int32_t i;
+            Peep* peep;
+            FOR_ALL_GUESTS (i, peep)
+            {
+                if (peep->state == PEEP_STATE_QUEUING_FRONT && peep->current_ride == 0)
+                {
+                    peep->RemoveFromQueue();
+                    peep->SetState(PEEP_STATE_FALLING);
+                    break;
+                }
+            }
+
+            // Now, swap the entrance and exit.
+            Ride* ride = get_ride(0);
+            auto entranceCoords = ride->stations[0].Exit;
+            auto exitCoords = ride->stations[0].Entrance;
+            ride->stations[0].Entrance = entranceCoords;
+            ride->stations[0].Exit = exitCoords;
+
+            auto entranceElement = map_get_ride_exit_element_at(
+                entranceCoords.x * 32, entranceCoords.y * 32, entranceCoords.z, false);
+            entranceElement->SetEntranceType(ENTRANCE_TYPE_RIDE_ENTRANCE);
+            auto exitElement = map_get_ride_entrance_element_at(exitCoords.x * 32, exitCoords.y * 32, exitCoords.z, false);
+            exitElement->SetEntranceType(ENTRANCE_TYPE_RIDE_EXIT);
+
+            // Trigger footpath update
+            footpath_queue_chain_reset();
+            footpath_connect_edges(
```

Comment:
Just a thought but does this spend money? GAME_COMMAND_FLAG_5 is normally passed in a fixing function as that tells it not to spend money (or network it).

Reply:
It only uses the flags but its not really a game command, it should be fine.

Reply:
Reconnecting footpath edges does not cost money.

> GAME_COMMAND_FLAG_5 is normally passed in a fixing function as that tells it not to spend money (or network it).

Shouldn't we rename that flag?

Reply:
I thought for a while that it should be renamed GAME_COMMAND_TRACK_PREVIEW. But now we use it for quite a few things not related to track previews. I guess its like GHOST in that it forces it to be client side only so that property of it should be documented but not part of the name. So maybe NO_COST would be appropriate but its a very oddly used command.